src/components/RateProratedMonthlyForm.js

```diff
@@ -28,15 +28,16 @@ const RateProratedMonthlyForm = (props) => {
 
     const [monthlyHoursInput, setMonthlyhoursInput] = useState(null)
     const [currentRateInput, setCurrentRateInput] = useState(null)
-    const [rateCurrency, setRateCurrency] = useState(clientCurrency)
+    const [rateCurrency, setRateCurrency] = useState(null)
     const [totalAmount, setTotalAmount] = useState(null)
     const [totalWeeks, setTotalWeeks] = useState(null)
     const [totalHours, setTotalHours] = useState(0)
 
     useEffect(() => {
-        if (rateCurrency) {
-            setCurrency(rateCurrency)
+        if (!rateCurrency) {
+            setRateCurrency(clientCurrency)
         }
+        setCurrency(rateCurrency)
```

Comment:
Not sure about these state structures dealing with `currency` maybe there is some refactoring/cleanup just leaving a note here but it's not clear exactly what's going on here, maybe even a comment to explain if this really is the best way to manage the currency state